### DISCUSSION TITLE
Ezcap269 has to be restarted everytime after a long idle state

### DISCUSSION THREAD

**Discussion by marcosvrs** [awawa-dev/HyperHDR#289] [Q&A] [answered]
I always have to restart my ezcap269 (Unplug and plug the USB) every time it stays in idle for a long time (e.g., 8h).
On HyperHDR I can see a typical “no signal” image coming from the ezcap269 at that moment.
Even the pass-through doesn't work! It's pretty annoying.

Is it normal? Is it a known bug? Any workaround to avoid me having to go behind my TV cabinet and unplug the USB cable?

**Comment by awawa-dev** [comment]
> Even the pass-through doesn't work!

Hardware, power supply or OS (USB power saving function) problem? USB grabber pass-through cannot be control by the software directly because it's independent.

**Reply by marcosvrs** [reply]
I tried everything...
Used `powertop` to disable the power save mode. It also shows that the ezcap is consuming 100% of the USB port power.
Tried different ports.
unbind and bind the USB port to check if the signal returns remotely.

I'm out of ideas...

**Reply by awawa-dev** [reply]
Even a very brief drop in power can hang an electronic device until it is restarted. Maybe Ezcap 269 is the problem? Did you try to update the firmware or connect to other PC with HyperHDR and leave it idle for few hours to eliminate current setup problem?

**Reply by marcosvrs** [reply]
Is there any firmware newer than `2018041303bM3BS-200417 TEST ONLY.bin`? Or any custom firmware?
It's not a bad idea to test it in another PC.
I'm afraid that my ezcap is broken. :(

**Reply by awawa-dev** [reply]
For 269 that's the latest firmware but for 269a on their page is ```TTL IN+2828_4K60-EZCAP_20201116.bin```. I don't know the difference between legacy 269 and newer 269a. There is completely no information/difference about 269a model and only 269 exists in the catalogue. Maybe ask Ezcap support about your problem (device suspends after few hours of idle state and doesn't wake up)  and if you can apply 269a firmware for your device?

As the last resort you can  probably try software power down&up RPI's USB hub to wake up ezcap 269:
https://stackoverflow.com/questions/59772765/how-to-turn-usb-port-power-on-and-off-in-raspberry-pi-4

**Reply by marcosvrs** [reply]
I'm running in an Intel NUC, so I don't think this rpi software would work. But thank you for the suggestion!
I'll check everything and also try to contact ezcap support (Have you experience on getting in contact with them? Do they normally help?)

**Reply by awawa-dev** [reply]
Yes, they reply fast and helped me with Ezcap 320 firmware problem (but why they released such broken firmware, didn't update the manual that mislead users to perform only partial update and later blame only them ....don't know)

**Reply by awawa-dev** [reply]
FYI: I updated my ezcap to 269A firmware and the capturing stopped to work. But I was able to return to 269 firmware and make it work again.

**Reply by marcosvrs** [reply]
I can confirm that the problem is between my Apple TV and ezcap.
My current setup:
Apple TV -> HT Receiver -> ezcap -> TV
I connected my Intel NUC into my HT and it works. I can see Intel NUCs output.
Once I switch the HDMI port from my HT to the Apple TV, my TV has no signal.
I restarted the Apple TV, the HT and the TV and it doesn’t change anything.
Only after I restart ezcap, everything works. 😪

**Reply by marcosvrs** [reply]
Can it be that the HDR output from my Apple TV is somehow not recognised after idle? Or Apple TV changes the output after idle. I have no idea.
Intel NUC output is raw 1080p.

**Reply by awawa-dev** [reply]
Don't know, Apple TV likes to enforce DV if it's possible and Ezcap can't handle it. Or maybe Apple TV sends some HDMI power saving command at idle and some bug in ezcap firmware causes to not wake it up later. I have completely no explanation why it works after Ezcap restart. Maybe new EDID handshake happens but it also should occur when you restart Apple/HT/TV.

**Reply by marcosvrs** [reply]
Do you know if I can control ezcap in any way remotely? Send a command to restart itself, or anything that can help to remotely work around that issue?

**Reply by awawa-dev** [reply]
No, the only software workaround  that comes to my mind is to restart host USB hub which should also restart all connected devices including Ezcap grabber. But it's probably possible only for Linux.

**Reply by marcosvrs** [reply]
Another test I made: Connected the Apple TV directly on my ezcap. It works flawlessly. Once I switch it back to my HT, it doesn't work anymore. Only once I restart the ezcap it works again.
So, it's the setup of Apple TV + HT Receiver + ezcap that doesn't work :(

**Reply by awawa-dev** [reply]
And it works in other way: first HT then Apple? Grabbers with loop have almost always problematic internal HDMI splitters. Sometimes it comes out when DHCP is in play or with long cascade of EDID handshakes (multiple AV devices in the path).

**Reply by awawa-dev** [reply]
Try to disable HDCP

**Comment by marcosvrs** [comment]
You are talking about which DHCP? CEC has DHCP? How can I disable that?

Kind Regards
Marcos Vinícius Rubido
________________________________

**Reply by awawa-dev** [reply]
Yes, misspelled HDCP

**Reply by awawa-dev** [reply]
On PC it's in the graphic card properties. Don't have Apple TV and don't know if it's possible for it.
My Denon AV can't disable HDCP on its own and it always forward HDCP content so it must be disabled on the source (or by using smart HDMI splitter).

**Comment by marcosvrs** [comment]
How to do that?

Kind Regards
Marcos Vinícius Rubido
________________________________

**Comment by marcosvrs** [comment]
I also have ordered a Intel NUC HDMI-CEC Adapter so I can debug the setup a bit better.

Kind Regards
Marcos Vinícius Rubido
________________________________

**Comment by marcosvrs** [comment]
Worse support ever:
> From: jacky@ezcap.com <jacky@ezcap.com>
Sent: Wednesday, June 22, 2022 2:18 AM
To: Marcos Vin?cius Rubido <marcosauda@hotmail.com>; Michael <michael@ezcap.com>
Subject: Re: Ezcap269 has to be restarted every time after a long idle state
> 
> Hi,
> 
> ezcap269 is our old solution, and the splliter and pass-through is not so good.
ezcap326 is the new solution, and it's better than ezcap269, maybe you can have a try.
> 
> Sorry, we have no enough informations on Apple TV and Yamaha HT Receiver.
> 
> Thanks and regards,
Jacky Wu 吴明俊
> 
> FORWARD VIDEO TECHNOLOGY CO., LTD. | 深圳市风云翔龙科技有限公司
w: [www.ezcap.com](https://nam12.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.ezcap.com%2F&data=05%7C01%7C%7C61126315f564415c781308da53e4c0e3%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637914539197857600%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=RSqkefmR56oZ9L7GEWcfYKGgQ%2FiT%2BSXXD2iHA2mhD5k%3D&reserved=0) | m: jacky@ezcap.com

**Comment by awawa-dev** [comment]
Well, at least they're honest. The Ezcap 269 was a good option, but then the MS2109 came out and as I said, grabbers with internal loop are almost always inferior than grabber+splitter solutions (even Ezcap 320 family). In fact, Apple TV HDCP causes some problems even with advanced external HDMI splitters like the Ezcoo EZ-SP12H2 that required a firmware update.

**Reply by marcosvrs** [reply]
Which grabber do you recommend? I was even considering buying the elgato 4k.

**Reply by awawa-dev** [reply]
I very doubt elgato will work for HDR and HyperHDR: Windows & proprietary codecs necessary are neccesery for that grabber. Currently I see two options for best quality: MS2109 (Rullz black USB2.0 version, some USB3.0 fakes are reported to provide worse quality) + FeinTech VSP01201. Or faster USB3.0 Ezcap 321 + FeinTech VSP01201 (my current setup).  External HDMI splitter is necessary to enforce HDR content and to remove HDCP for these grabbers. Ezcap 320 (and probably similar 326) is all in one solution for HDR but the quality is worse and it gives up when receives HDCP protected signal anyway.

**Comment by marcosvrs** [comment]
The Intel NUC HDMI-CEC Adapter arrived, and I confirmed what you already told me: ezcap is messing up with the HDCP of my setup. I can't see the ezcap and not even my TV in the network. Also, my TV isn't syncing the standby status in my setup, and I need to have the TV remote every time to just turn it on/off.
Therefore, I decided to go with your suggestion and add a splitter to my setup (I'll not buy the new ezcap as it looks like it doesn't worth it).

But I have one question: What is the difference between the **FeinTech VSP01201** and **FeinTech VSP01204**? As it looks like they work the same, but the **FeinTech VSP01204** doesn't need power.
Will it really work with **FeinTech VSP01204** or I must use the **FeinTech VSP01201** to have CEC/HDCP to work with my TV/ezcap?

**Comment by awawa-dev** [comment, marked as answer]
I would avoid  FeinTech VSP01204 just for one reason: it doesn't have the EDID switch and it could cause very serious incompatibility for some setups. I guess it always works in MIXED mode known from VSP01201. So if you connect HDR capable TV and non-HDR capable grabber it would force the video source to drop HDR and force SDR. So you need VSP01201 COPY mode for Ezcap 269 or MS2109 with TV connected to VSP01201 HDMI 1 port and grabber to HDMI 2 port. BTW Rullz black USB2.0 is very cheap around 10$ and the quality is much better than Ezcap 269 (latency is similar). My VSP01201 has always constant power supply from PC USB port.

**Reply by marcosvrs** [reply]
Is this the genuine one?
https://de.aliexpress.com/item/4000927743306.html?spm=a2g0o.seodetail.topbuy.1.2762a9e1kkJI5E

**Reply by awawa-dev** [reply]
That black one USB2.0 version looks like my Rullz, but it's Aliexpress so you never know what you will receive. Probably the more orders/user reviews the better. Just don't order USB3.0/1060p60 coffee color version.

**Reply by marcosvrs** [reply]
What about this one?
https://www.berrybase.de/usb-hdmi-video-grabber-1080p#
Also using MS2109, the only thing is that I don't know if it has the same issue as the USB3 coffee color version that you mentioned.

**Reply by marcosvrs** [reply]
Update: I just bought it. For €9 it is worth to try it out :)

**Comment by marcosvrs** [comment]
My FeinTech VSP01201 just arrived.
On Copy mode, my ezcap269 doesn't give any image. And on mixed, it doesn't forward HDR10 to the TV :(

**Reply by marcosvrs** [reply]
But at least the CEC with my TV works like a charm :)
RIP TV remote.

**Reply by marcosvrs** [reply]
Found the issue.
The Apple TV was delivering 4:2:2. I changed to 4:2:0 and now ezcap works.

**Reply by awawa-dev** [reply]
Probably 4:2:2 is too much for Ezcap 269. Sometimes it depends also on the overall video bandwidth for example may work for 1080p60 but not for 4k60

**Reply by marcosvrs** [reply]
Sad, but TBH I can't see the difference between 4:2:0 and 4:2:2 on my TV. So, I can live with the chroma like that.
